contracts/InsuranceArbitrator.sol

```diff
@@ -75,40 +70,21 @@ contract InsuranceArbitrator {
         address indexed insurer,
         string insuredEvent,
         address indexed insuredAddress,
-        IERC20 currency,
-        uint256 insuredAmount
-    );
-    event ClaimSubmitted(
-        uint256 claimTimestamp,
-        bytes32 indexed policyId,
-        string insuredEvent,
-        address indexed insuredAddress,
-        IERC20 currency,
-        uint256 insuredAmount
-    );
-    event ClaimAccepted(
-        uint256 claimTimestamp,
-        bytes32 indexed policyId,
-        string insuredEvent,
-        address indexed insuredAddress,
-        IERC20 currency,
-        uint256 insuredAmount
-    );
-    event ClaimRejected(
-        uint256 claimTimestamp,
-        bytes32 indexed policyId,
-        string insuredEvent,
-        address indexed insuredAddress,
-        IERC20 currency,
         uint256 insuredAmount
     );
+    event ClaimSubmitted(uint256 claimTimestamp, bytes32 indexed claimId, bytes32 indexed policyId);
+    event ClaimAccepted(bytes32 indexed claimId, bytes32 indexed policyId);
+    event ClaimRejected(bytes32 indexed claimId, bytes32 indexed policyId);
 
     /**
      * @notice Construct the InsuranceArbitrator
      * @param _finderAddress DVM finder to find other UMA ecosystem contracts.
+     * @param _currency denomination token for insurance coverage and bonding.
      */
-    constructor(address _finderAddress) {
+    constructor(address _finderAddress, address _currency) {
         finder = FinderInterface(_finderAddress);
+        currency = IERC20(_currency);
```

Comment:
We do not check currency being on whitelist anymore assuming the contract is deployed with approved currency.

Note that also with custom currency per insurance this was not perfect if currency was removed from whitelist after insurance is issued.